web/spacewalk-web.spec

```diff
@@ -40,18 +40,20 @@ Group:          Applications/Internet
 Version:        4.2.4
 Release:        1%{?dist}
 Url:            https://github.com/uyuni-project/uyuni
-Source0:        https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
+Source0:        https://github.com/uyuni-project/uyuni/archive/%{name}-%{version}.tar.gz
 BuildRoot:      %{_tmppath}/%{name}-%{version}-build
 BuildArch:      noarch
 Requires(pre):  uyuni-base-common
 BuildRequires:  uyuni-base-common
 BuildRequires:  perl(ExtUtils::MakeMaker)
 BuildRequires:  susemanager-nodejs-sdk-devel
+BuildRequires:  nodejs-packaging
 
 %if 0%{?suse_version}
 BuildRequires:  apache2
-BuildRequires:  nodejs-packaging
-
+BuildRequires:  nodejs10
+%else
+BuildRequires:  nodejs
```

Comment:
Just a comment from my side, shouldn't this be `BuildRequires:  nodejs = 10` or similar, so we also stick to a given nodejs version for CentOS?

Reply:
If I recall correctly, we already previously tried this, either on master or on 4.1 and there was a problem with it, but I can't recall what the specific problem was. I can't find a paper trail right now.

Reply:
I can certainly try it as it is now in an %else clause.

Reply:
This would work out to 
BuildRequires:    nodejs >= 1:10, nodejs <= 1:11
This looks very RHEL8 specific. Not sure if we would really want it like that.
Either way, putting line 56 to the above builds on Leap and CentOS 8.

Reply:
I found the old discussion about version numbers as well, from [our internal chat](https://chat.suse.de/channel/galaxy-naica?msg=6qcZnem8xkxjwG9jZ):  

> Our nodejs packages for not provide the simple version number.  
> ```
> nothing provides nodejs = 10 needed by nodejs-packaging
> (got version 8.17.0 provided by nodejs8)
> (got version 12.18.4 provided by nodejs12)
> (got version 14.15.0 provided by nodejs14)
> (got version 10.22.1 provided by nodejs10)```

With a link to https://build.opensuse.org/package/view_file/openSUSE:Leap:15.2:Update/nodejs10/nodejs10.spec?expand=1 line 284

Reply:
This might be because you put `nodejs = 10`. That won't work (as far as I understand. You have to put `>=` and `<`. So I would expect` Requires nodejs >= 10, nodejs < 11` to work (unless you call the version `1:10` and `1:11` like in the RHEL8 case...).